### ISSUE TITLE
canu is generating a very large index file

### ISSUE THREAD

**Issue by tangerzhang** [marbl/canu#302]
Hi,
My canu is running ovStoreIndexer step (asm.ovlStore.BUILDING/scripts/3-index.sh) and is generating a very large index file (asm.ovlStore.BUILDING/index, current 42TB).
The genome size is only 450m. 
Is that normal, or something wrong?
My config file is listed:
useGrid=1
gridOptions=-S /bin/bash -q all.q -l mem_free=60g
gridEngineThreadsOption=-pe mpi THREADS
gridEngineMemoryOption=-l mem_total=MEMORY
corOutCoverage=120
ovbMemory=8g
maxMemory=500g
maxThreads=48
ovsMemory=8-500g
ovsThreads=4

**Comment by brianwalenz** [comment]
Definitely wrong.  You can kill it and remove the index file.  All this is doing is merging the various index files in the `asm.ovlStore.BUILDING` directory, so it should be relatively tiny.

If the `asm.ovlStore.BUILDING/logs/3-index.err` file exists, please paste it in.  I suspect I'll need to add some debugging and have you rerun.  Fortunately, just the 3-index.sh step needs to be rerun.

**Comment by tangerzhang** [comment]
Thanks for your response.
Below is the information of 3-index.err and please let me know when you fix it. Thanks very much.


Processing '/share/bioinfo/zhaoq/project//1_canu/asm80X/correction/asm.ovlStore.BUILDING/index'
  Now finished with fragments 1 to 374268 -- 202097028 overlaps.
Processing '/share/bioinfo/zhaoq/project//1_canu/asm80X/correction/asm.ovlStore.BUILDING/0001.index'
  Now finished with fragments 1 to 600660 -- 404199022 overlaps.
Processing '/share/bioinfo/zhaoq/project//1_canu/asm80X/correction/asm.ovlStore.BUILDING/0002.index'
  Now finished with fragments 1 to 813580 -- 606287472 overlaps.
Processing '/share/bioinfo/zhaoq/project//1_canu/asm80X/correction/asm.ovlStore.BUILDING/0003.index'
  Now finished with fragments 1 to 1012795 -- 808375001 overlaps.
Processing '/share/bioinfo/zhaoq/project//1_canu/asm80X/correction/asm.ovlStore.BUILDING/0004.index'
  Now finished with fragments 1 to 1211170 -- 1010480763 overlaps.
Processing '/share/bioinfo/zhaoq/project//1_canu/asm80X/correction/asm.ovlStore.BUILDING/0005.index'
  Now finished with fragments 1 to 1415235 -- 1212586675 overlaps.
Processing '/share/bioinfo/zhaoq/project//1_canu/asm80X/correction/asm.ovlStore.BUILDING/0006.index'
  Adding empty records for fragments 1415236 to 1415236

**Comment by Mullus** [comment]
Hi, similar issue here.

When trimming reads (162000) genome size 350m, Canu filled up all my disk space (544GB). Here is the log file.

Thank you in advance for any comments.

-- Overlap store sorter finished.
----------------------------------------
-- Starting command on Sun Nov 27 21:10:28 2016 with 544.358 GB free disk space

    /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore.BUILDING/scripts/3-index.sh \
    > /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore.BUILDING/logs/3-index.err 2>&1

-- Finished on Sun Nov 27 21:41:32 2016 (1864 seconds) with 0 GB free disk space  !!! WARNING !!!
----------------------------------------
--
-- Overlap store '/data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore' successfully constructed.
--
-- Purged 0.231 GB in 45 overlap output files and 1 directories.
----------------------------------------
-- Starting command on Sun Nov 27 21:41:32 2016 with 0.231 GB free disk space

    /Software/canu/Linux-amd64/bin/ovStoreStats \
     -G /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.gkpStore \
     -O /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore \
     -o /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore \
     > /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore.summary.err 2>&1

-- Finished on Sun Nov 27 21:41:32 2016 (lickety-split) with 0.231 GB free disk space  !!! WARNING !!!
----------------------------------------
ERROR:
ERROR:  Failed with exit code 1.  (rc=256)
ERROR:
--
-- WARNING: failed to generate statistics for the overlap store; no summary will appear in HTML output.
--
----------------------------------------
-- Overlap store '/data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore' statistics not available.
----------------------------------------
-- Starting command on Sun Nov 27 21:41:32 2016 with 0.231 GB free disk space

    /Software/canu/Linux-amd64/bin/trimReads \
      -G  /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.gkpStore \
      -O  /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore \
      -Co /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/3-overlapbasedtrimming/Trim.1.trimReads.clear \
      -e  0.045 \
      -minlength 1000 \
      -ol 1 \
      -oc 1 \
      -o  /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/3-overlapbasedtrimming/Trim.1.trimReads \
    >     /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/3-overlapbasedtrimming/Trim.1.trimReads.err 2>&1

-- Finished on Sun Nov 27 21:41:32 2016 (lickety-split) with 0.231 GB free disk space  !!! WARNING !!!
----------------------------------------
ERROR:
ERROR:  Failed with exit code 1.  (rc=256)
ERROR:
--
-- WARNING: failed to generate statistics for the overlap store; no summary will appear in HTML output.
--
----------------------------------------
-- Overlap store '/data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore' statistics not available.
----------------------------------------
-- Starting command on Sun Nov 27 21:41:32 2016 with 0.231 GB free disk space

    /Software/canu/Linux-amd64/bin/trimReads \
      -G  /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.gkpStore \
      -O  /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore \
      -Co /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/3-overlapbasedtrimming/Trim.1.trimReads.clear \
      -e  0.045 \
      -minlength 1000 \
      -ol 1 \
      -oc 1 \
      -o  /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/3-overlapbasedtrimming/Trim.1.trimReads \
    >     /data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/3-overlapbasedtrimming/Trim.1.trimReads.err 2>&1
-- Finished on Sun Nov 27 21:41:32 2016 (lickety-split) with 0.231 GB free disk space  !!! WARNING !!!
----------------------------------------
ERROR:
ERROR:  Failed with exit code 1.  (rc=256)
ERROR:
================================================================================
Please panic.  canu failed, and it shouldn't have.

Stack trace:

 at /Software/canu/Linux-amd64/bin/lib/canu/Execution.pm line 1305.
        canu::Execution::caFailure("trimReads failed", "/data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/3-o"...) called at /Software/canu/Linux-amd64/bin/lib/canu/OverlapBasedTrimming.pm line 80
        canu::OverlapBasedTrimming::trimReads("/data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2", "Trim") called at /Software/canu/Linux-amd64/bin/canu line 506

Last few lines of the relevant log file (/data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/3-overlapbasedtrimming/Trim.1.trimReads.err):

ERROR: directory '/data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore' is not an overlapStore; didn't find file '/data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore/info': No such file or directory
ERROR:  failed to intiialize ovStore '/data/galarza/Canu_Formica/Corr1NS-ER-035/Trim-2/trimming/Trim.ovlStore'.

canu failed with 'trimReads failed'.

**Comment by brianwalenz** [comment]
Tanger- It's fixed.  It was (embarrassingly) definitely trying to fill up your disk for no good reason.  You can restart canu and It should pick up where it left off.  Or run the 3-index.sh script by hand then restart.

Mr and/or Ms Mullus - yours is harder to restart, because canu didn't stop after filling your disk.  Paste in the contents of `Trim.ovlStore` so I can make sure it actually failed correctly.  I'm worried that it removed the intermediate files already, and so the only thing you can do is restart from scratch.

**Comment by tangerzhang** [comment]
Thanks, Brian.

2016-11-29 15:16 GMT+08:00 Brian Walenz <notifications@github.com>:

**Comment by Mullus** [comment]
Thank you Brian. Unfortunately my system admin removed the entire run directory as it was using all disk space (he was not too happy). Should I then download and reinstall Canu before starting a new assembly?.

Cheers

**Comment by brianwalenz** [comment]
You'll most definitely want to update Canu before starting again!  Open a new issue if you have trouble again.
